### PR TITLE
Add license key editing form.

### PR DESCRIPTION
**Pull request by squidsoup**
## Done
Added form for editing/updating license keys.

## QA
1. Ensure you have a fake windows image uploaded (if not see _Creating a fake windows image_).
2. Visit /MAAS/settings/license-keys in maas-ui, click the _add license key_ button, and create a new license key (windows license keys are in the form `XXXXX-XXXXX-XXXXX-XXXXX-XXXXX`).
3. Return to the license key list, click the edit button next to your key.
4. Ensure the edit license key form displays the correct key data. 
5. Edit your key, and click "update license key".
6. The form should redirect you to the licence key list with a success message.
7. Edit your key again, and ensure the new key is reflected in the form.

## Creating a fake windows image
1. Ensure you have an amd64 ubuntu image downloaded (via http://maas.local:5240/MAAS/#/images), this is required to populate architecture for the following steps.
2. Create a fake windows image with `dd if=/dev/zero of=windows-dd bs=512 count=10000`
3. Upload the image you've created with `maas {profile} boot-resources create name=windows/win2012 title="Windows Server 2012" architecture=amd64/generic filetype=ddtgz content@=windows-dd`. If you haven't used the maas cli before, you'll need to maas login first and create a profile.

### CODE REVIEW

File: ui/src/app/settings/views/LicenseKeys/LicenseKeyForm/LicenseKeyForm.js [hunk 1]
```diff
@@ -57,16 +59,20 @@ export const LicenseKeyForm = () => {
     return <Redirect to="/settings/license-keys" />;
   }
 
+  const title = licenseKey ? "Update license key" : "Add license key";
```
Comment (huwshimi) [inline ui/src/app/settings/views/LicenseKeys/LicenseKeyForm/LicenseKeyForm.js]: You could move this higher and reuse for `useWindowTitle`.

File: ui/src/app/settings/views/LicenseKeys/LicenseKeyForm/LicenseKeyForm.js [hunk 2]
```diff
@@ -32,10 +32,12 @@ export const LicenseKeyForm = () => {
 
   useWindowTitle("Add license key");
```
Comment (huwshimi) [inline ui/src/app/settings/views/LicenseKeys/LicenseKeyForm/LicenseKeyForm.js]: Needs to use the title from below for editing.

**Comment by huwshimi** [comment]
As discussed the API is sometimes returning a 400 but this code handles the errors nicely.

**Comment by squidsoup** [comment]
Have filed a bug for [License key API erroneously reports "Invalid license key"](https://bugs.launchpad.net/maas/+bug/1848097), but merging this as this issue is almost certainly on the API side.

**Pull request merged**